### PR TITLE
Rent model

### PR DESCRIPTION
**Pull request by floresv**
### Summary
Includes:
*  Rent migration
* Rent model, with validations and relations
* Rent model tests

### Trello Card
https://trello.com/b/oicSvuOV/training-rails-javier-flores

### CODE REVIEW

File: app/models/rent.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+class Rent < ApplicationRecord
+  validates :user_id, :book_id, :from, :to, presence: true
```
Comment (leamotta) [inline app/models/rent.rb]: Are you sure you want to validate just the id here or the whole entity? To understand the difference you could read this for example
https://forum.upcase.com/t/validate-presence-of-user-id-or-user/6168/4

File: app/models/rent.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  validate :from_cannot_be_greater_than_to, if: proc { |r| r.from.present? && r.to.present? }
```
Comment (leamotta) [inline app/models/rent.rb]: you could also write this as 
`if: -> { from.present? && to.present? }`
you don't need to change, it's just an FYI

File: app/models/user.rb [hunk 3]
```diff
@@ -1,7 +1,9 @@
 class User < ApplicationRecord
   include DeviseTokenAuth::Concerns::User
-  # Include default devise modules. Others available are:
-  # :confirmable, :lockable, :timeoutable and :omniauthable
+
+  has_many :rents, dependent: :destroy
```
Comment (leamotta) [inline app/models/user.rb:4]: You can do the same for a `Book` :)

File: config/environments/development.rb [hunk 4]
```diff
@@ -31,6 +31,8 @@
 
   config.action_mailer.perform_caching = false
 
+  config.action_mailer.default_url_options = { :host => "localhost" }
```
Comment (leamotta) [inline config/environments/development.rb:34]: Is this in this PR for something in particular?

File: spec/models/rent_spec.rb [hunk 5]
```diff
@@ -0,0 +1,13 @@
+require 'rails_helper'
+
+describe Rent do
+  subject(:rent) { create(:rent) }
+
+  it { is_expected.to validate_presence_of :user_id }
+  it { is_expected.to validate_presence_of :book_id }
```
Comment (leamotta) [inline spec/models/rent_spec.rb]: same here with what I said in the model validation
Also, parenthesis are sad that you forgot about them haha

File: spec/models/rent_spec.rb [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+  it { is_expected.to validate_presence_of :from }
+  it { is_expected.to validate_presence_of :to }
+  it { is_expected.to satisfy('from cannot be greater than to') { |rent| rent.from <= rent.to } }
```
Comment (leamotta) [inline spec/models/rent_spec.rb]: 🤔  I've never used `satisfy`, I'll have to look it up.
We usually do something like `it { expect(rent.from).to be <= rent.to }`

**Review by leamotta — COMMENTED** [review]
Some comments, let me know what you think

File: app/models/rent.rb (hunk 1 above)
Comment (floresv) [reply app/models/rent.rb]: Yes, I will validate the whole entity.

File: config/environments/development.rb (hunk 4 above)
Comment (floresv) [reply config/environments/development.rb:34]: It's required when I tried execute Rent Tests with User devise model.

File: spec/models/rent_spec.rb (hunk 5 above)
Comment (floresv) [reply spec/models/rent_spec.rb]: Sorry, I have the habit of don't use parentheses.

File: spec/models/rent_spec.rb (hunk 6 above)
Comment (floresv) [reply spec/models/rent_spec.rb]: Oks, I will try do it like that.

File: app/models/user.rb [hunk 7]
```diff
@@ -2,6 +2,7 @@ class User < ApplicationRecord
   include DeviseTokenAuth::Concerns::User
 
   has_many :rents, dependent: :destroy
+  has_many :books, dependent: :destroy
```
Comment (leamotta) [inline app/models/user.rb]: What I meant here is that a `Book` entity could have many `rents`, but this is fine. The only thing you are missing is the `through` option. You need to specify from where does the user obtain the books, in this case it's through rents.
Try it out via rails console creating a User, a book, some rents and then doing something like `user.books` to see if it works or not.

Also you can populate the `seeds.rb` file so you don't have to create those examples over and over again to test

Comment (floresv) [reply app/models/user.rb]: That's True :man_facepalming: Sorry!

**Review by leamotta — APPROVED** [review]

**Pull request merged**